iothub/device/src/Authentication/IotHubConnectionCredentials.cs

```diff
@@ -25,6 +26,9 @@ public sealed class IotHubConnectionCredentials : IConnectionCredentials
         /// <exception cref="ArgumentException"><paramref name="iotHubHostName"/> or device Id are an empty string or consist only of white-space characters.</exception>
         /// <exception cref="ArgumentException">Neither shared access key, shared access signature or X509 certificates were presented for authentication.</exception>
         /// <exception cref="ArgumentException">Either shared access key or shared access signature where presented together with X509 certificates for authentication.</exception>
+        /// <exception cref="ArgumentException"><see cref="DeviceAuthenticationWithX509Certificate"/> is used but <see cref="DeviceAuthenticationWithX509Certificate.Certificate"/> is null.</exception>
```

Comment:
Can that exception be thrown when initalizing `DeviceAuthenticationWithX509Certificate` instead?

Reply:
That's a good idea, yeah